core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -0,0 +1,162 @@
+            metadataCache: MetadataCache,
+            time: Time): TransactionMarkerChannelManager = {
+
+    val channel = new TransactionMarkerChannel(config.interBrokerListenerName, metadataCache)
+    val sendThread: InterBrokerSendThread = {
+      val threadName = "TxnMarkerSenderThread-" + config.brokerId
+      val channelBuilder = ChannelBuilders.clientChannelBuilder(
+        config.interBrokerSecurityProtocol,
+        JaasContext.Type.SERVER,
+        config,
+        config.interBrokerListenerName,
+        config.saslMechanismInterBrokerProtocol,
+        config.saslInterBrokerHandshakeRequestEnable
+      )
+      val selector = new Selector(
+        NetworkReceive.UNLIMITED,
+        config.connectionsMaxIdleMs,
+        metrics,
+        time,
+        "replica-fetcher",
+        Map("broker-id" -> config.brokerId.toString).asJava,
+        false,
+        channelBuilder
+      )
+      val networkClient = new NetworkClient(
+        selector,
+        new ManualMetadataUpdater(),
+        threadName,
+        1,
+        0,
+        Selectable.USE_DEFAULT_BUFFER_SIZE,
+        config.socketReceiveBufferBytes,
+        config.requestTimeoutMs,
+        time,
+        false,
+        new ApiVersions
+      )
+
+      new InterBrokerSendThread(threadName, networkClient, requestGenerator(channel), time)
+    }
+
+    new TransactionMarkerChannelManager(config,
+      metadataCache,
+      new KafkaScheduler(1, "transaction-marker-channel-manager"),
+      sendThread,
+      channel)
+  }
+
+
+  private[transaction] def requestGenerator(transactionMarkerChannel: TransactionMarkerChannel): () => immutable.Map[Node, RequestAndCompletionHandler] = {
+    def generateRequests(): immutable.Map[Node, RequestAndCompletionHandler] = {
+      transactionMarkerChannel.brokerStateMap.flatMap {case (brokerId: Int, destAndMarkerQueue: DestinationBrokerAndQueuedMarkers) =>
+        val markersToSend: java.util.List[TxnMarkerEntry] = new util.ArrayList[TxnMarkerEntry] ()
+        destAndMarkerQueue.markersQueue.drainTo (markersToSend)
+        val requestCompletionHandler = new TransactionMarkerRequestCompletionHandler(transactionMarkerChannel, markersToSend, brokerId)
+        if (markersToSend.isEmpty)
+          None
+        else
+          Some ((destAndMarkerQueue.destBrokerNode, RequestAndCompletionHandler (new WriteTxnMarkerRequest.Builder (markersToSend), requestCompletionHandler) ) )
+      }.toMap
+    }
+    generateRequests
+  }
+}
+
+
+
+class TransactionMarkerChannelManager(config: KafkaConfig,
+                                      metadataCache: MetadataCache,
+                                      scheduler: Scheduler,
+                                      interBrokerSendThread: InterBrokerSendThread,
+                                      transactionMarkerChannel: TransactionMarkerChannel) extends Logging {
+
+  type WriteTxnMarkerCallback = Errors => Unit
+
+  // TODO: Config for how often this runs?
```

Comment:
I think we shouldn't add a config right now, but pick a reasonable default instead. That's the approach we have taken to other such parameters as well.